### PR TITLE
[DM Feature] Indent nested loop source nodes

### PR DESCRIPTION
**Pull request by DanielleCogs**
Three main changes with this PR:
1. Parents of repeating items are added when child is added, not when connection is made (still kept code to add back parent if child is connected, in case it was removed)
2. In-between ancestors are added if the repeating element has a higher ancestor on the canvas
3. When the source canvas changes, all nodes sizes are recalculated

### CODE REVIEW

File: libs/data-mapper/src/lib/utils/Schema.Utils.ts [hunk 1]
```diff
@@ -47,6 +55,35 @@ export const parsePropertiesIntoNodeProperties = (propertiesString: string): Sch
   return [];
 };
 
+export const setWidthForSourceNodes = (sourceNodes: SourceSchemaNodeExtended[]) => {
```
Comment (DanielleCogs) [inline libs/data-mapper/src/lib/utils/Schema.Utils.ts:58]: this is the more efficient version that needs to be refactored

File: libs/data-mapper/src/lib/utils/ReactFlow.Util.ts [hunk 2]
```diff
@@ -178,6 +178,26 @@ const convertSourceToReactFlowParentAndChildNodes = (
     return reactFlowNodes;
   }
 
+  const sourceNodesCopy = [...combinedSourceSchemaNodes];
+  sourceNodesCopy.filter((node) => node.nodeProperties.includes(SchemaNodeProperty.Repeating));
+  const sourceKeySet: Set<string> = new Set();
+  sourceNodesCopy.forEach((node) => sourceKeySet.add(node.key));
+  const widthDict: Map<string, number> = new Map<string, number>();
+  let maxSize = 200;
```
Comment (refortie) [inline libs/data-mapper/src/lib/utils/ReactFlow.Util.ts:186]: Nit: Maybe add some of these numbers to constants and then reference them here?

File: libs/data-mapper/src/lib/utils/__test__/DataMapUtils.spec.ts [hunk 3]
```diff
@@ -20,6 +23,35 @@ import {
 } from '../__mocks__';
 
 describe('utils/DataMap', () => {
+  describe('addAncestorNodesToCanvas', () => {
+    const sourceSchema: Schema = layeredLoopSourceMockSchema;
+    const extendedSourceSchema: SchemaExtended = convertSchemaToSchemaExtended(sourceSchema);
+    const flattenedSchema = flattenSchemaIntoDictionary(extendedSourceSchema, SchemaType.Source);
+    it('includes direct parent', () => {
```
Comment (refortie) [inline libs/data-mapper/src/lib/utils/__test__/DataMapUtils.spec.ts:30]: Nit: some new lines here and below to break the `it()` from each other

**Review by refortie — APPROVED** [review]

File: libs/data-mapper/src/lib/models/Schema.ts [hunk 4]
```diff
@@ -55,11 +55,13 @@ export interface SchemaExtended extends Schema {
   schemaTreeRoot: SchemaNodeExtended;
 }
 
+export type SourceSchemaNodeExtended = SchemaNodeExtended & { width?: number };
 export interface SchemaNodeExtended extends SchemaNode {
   children: SchemaNodeExtended[];
   nodeProperties: SchemaNodeProperty[];
   // Inclusive of the current node
   pathToRoot: PathItem[];
+  width?: number;
```
[the remark this answers is not in the document]
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/models/Schema.ts:63]: done!

File: libs/data-mapper/src/lib/utils/ReactFlow.Util.ts (hunk 2 above)
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/utils/ReactFlow.Util.ts]: done!

File: libs/data-mapper/src/lib/utils/__test__/DataMapUtils.spec.ts (hunk 3 above)
Comment (DanielleCogs) [reply libs/data-mapper/src/lib/utils/__test__/DataMapUtils.spec.ts:31]: done!

**Pull request merged**